diff --git a/conan/tools/cmake/utils.py b/conan/tools/cmake/utils.py
index 0df8e5ff3..c944377e1 100644
--- a/conan/tools/cmake/utils.py
+++ b/conan/tools/cmake/utils.py
@@ -52,6 +52,9 @@ def cmake_escape_value(v):
             if char == '\\' and i + 1 < len(s) and s[i+1] in r'\$"':
                 yield s[i:i+2]  # Yield the existing pair
                 next(iterable)   # Skip the next char in the loop
+            elif char == '$' and i + 1 < len(s) and s[i+1] == '{':
+                # CMake variable ${...}: leave $ unescaped, keep moving
+                yield char
             elif char in r'\$"':
                 yield '\\' + char  # Escape the lonely char
             else:
diff --git a/test/integration/toolchains/cmake/test_cmaketoolchain.py b/test/integration/toolchains/cmake/test_cmaketoolchain.py
index 67e71c5db..6fd3ae29d 100644
--- a/test/integration/toolchains/cmake/test_cmaketoolchain.py
+++ b/test/integration/toolchains/cmake/test_cmaketoolchain.py
@@ -850,6 +850,7 @@ def test_variables_escaping():
             def generate(self):
                 toolchain = CMakeToolchain(self)
                 toolchain.variables["FOO"] = r"D:\new\thing\path"
+                toolchain.variables["CMAKE_Fortran_FLAGS_INIT"] = "${CMAKE_C_FLAGS_INIT}"
                 toolchain.variables.release["BAR"] = r"C:\new\thing\path"
                 toolchain.generate()
         """)
@@ -857,6 +858,7 @@ def test_variables_escaping():
     client.run("install . --name=mylib --version=1.0")
 
     toolchain = client.load("conan_toolchain.cmake")
+    assert 'set(CMAKE_Fortran_FLAGS_INIT "${CMAKE_C_FLAGS_INIT}"' in toolchain
     assert r'set(FOO "D:\\new\\thing\\path" CACHE STRING' in toolchain
     assert r'set(CONAN_DEF_releaseBAR "C:\\new\\thing\\path")' in toolchain
 
diff --git a/test/unittests/tools/cmake/test_cmake_test.py b/test/unittests/tools/cmake/test_cmake_test.py
index d507557f3..3d10ce8bc 100644
--- a/test/unittests/tools/cmake/test_cmake_test.py
+++ b/test/unittests/tools/cmake/test_cmake_test.py
@@ -99,6 +99,8 @@ def test_run_ctest():
     (r'TrailingSlash\\', r'TrailingSlash\\'),
     (r"Double\\Slash", r"Double\\Slash"),
     (r"", r""),  # Empty string
+    (r"${CMAKE_C_FLAGS_INIT}", r"${CMAKE_C_FLAGS_INIT}"),  # CMake variables no escaped
+    (r"${CMAKE_${LANG}_FLAGS_INIT}", r"${CMAKE_${LANG}_FLAGS_INIT}"),
 ])
 def test_cmake_escape_correctness(input_str, expected):
     escaped = cmake_escape_value(input_str)
